src/blocks/content-helper/components/related-top-post-list.tsx

```diff
@@ -26,8 +27,17 @@ function RelatedTopPostList() {
 	useEffect( () => {
 		const fetchPosts = async ( retries: number ) => {
 			ContentHelperProvider.getRelatedTopPosts()
-				.then( ( result ) => {
-					setPosts( result.posts );
+				.then( ( result ): void => {
+					const mappedPosts: RelatedTopPostData[] = result.posts.map(
+						( post: RelatedTopPostData ): RelatedTopPostData => (
+							{
+								...post,
+								date: dayjs( post.date ).format( 'MMM D, YYYY' ),
```

Comment:
This is a very US-centric date format. Is there an option to use either the site format, or a pre-existing user-defined / locale format?

Reply:
We have a very limited space on sidebar so seems like using anything else than this can move things to second line which will not look from UI perspective .... I think for now its better to stick with this and in future we can look into this if there are any feedback from customer to change this.

BTW for different languages can I use `__` function here? 🤔

Reply:
Hey @mehmoodak, just FYI, I had a similar issue and here is the solution I came up with:

https://github.com/Parsely/wp-parsely/blob/a87b21b95c6435206c479ad12373c1e5f9996e6d/src/blocks/content-helper/current-post-details/component.tsx#L106-L110

This outputs a short date, taking into account the current language of the page. It also does not introduce any external dependencies. And it is customizable so you can format the date the way you want. You can give it a try if you want.

Reply:
> We have a very limited space on sidebar so seems like using anything else than this can move things to second line which will not look from UI perspective

Let me introduce you to those using German as their locale in wp-admin... :D 

(German is notorious for using longer words without breaks, so if you're short of space for anything in English, it needs a re-think.)

Reply:
We should strive to keep things short (which promotes our best design and may work on most languages). For languages that will naturally be too long, there should be a fallback that is not perfect, but is at least viewable. If this is not possible (it looks bad when long), then maybe we would need feedback from @garrettvercoe.

Reply:
Thanks @acicovic for sharing the example. Implemented it and removed the `dayjs` dependency which is now extra.

As far as long names are concerned the UI will not break so I think we are good here :)